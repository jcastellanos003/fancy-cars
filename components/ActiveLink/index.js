import { useRouter } from 'next/router'
import Link from 'next/link';

const ActiveLink = ({ children, ...props }) => {
  const router = useRouter();
  const child = React.Children.only(children);
  
  return (
    <Link {...props}>
      {React.cloneElement(child, { active: router.pathname === props.href, theme: props.theme })}
    </Link>
  )
};

export default ActiveLink;
