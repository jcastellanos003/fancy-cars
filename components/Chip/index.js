import PropTypes from 'prop-types';

import * as components from './styles';

export const Chip = ({ name, onClose }) => (
  <components.StyledChip>
    <span>{name}</span> <components.CloseIcon onClick={onClose} />
  </components.StyledChip>
);

Chip.propTypes = {
  name: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired
};
