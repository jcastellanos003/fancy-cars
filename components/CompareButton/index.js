import * as components from './styles';

export const CompareButton = () => (
  <components.StyledCompareButton>
    <components.CompareButtonIcon />
  </components.StyledCompareButton>
);
