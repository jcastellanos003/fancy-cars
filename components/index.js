export * from './ActiveLink';
export * from './Nav';
export * from './Meta';
export * from './SearchBox';
export * from './Card';
export * from './VehicleCard';
export * from './BackButton';
export * from './PageHeader';
export * from './Chip';
export * from './CompareButton';
export * from './PageTitle';
export * from './PriceLabel';
