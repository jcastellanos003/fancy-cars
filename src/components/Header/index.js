import { Nav } from '../Nav';
import * as styled from './styles';

export const Header = () => (
  <section>

    <styled.CustomHeader>
      <styled.Logo src="/logo.png"/>
    </styled.CustomHeader>

    <Nav />

  </section>
);
