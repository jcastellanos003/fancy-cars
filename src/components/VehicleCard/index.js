import PropTypes from 'prop-types';

import { Card, CardHeader, CardContent  } from '../Card';
import { VehicleImage } from './styles';

export const VehicleCard = ({
    image,
    size,
    children
  }) => (
    <section>
      <Card size={size}>

        <CardHeader>
          <VehicleImage src={image} />
        </CardHeader>

        <CardContent>
          {children}
        </CardContent>

      </Card>
    </section>
);

VehicleCard.propTypes = {
  size: PropTypes.oneOf(['medium', 'large']).isRequired,
  image: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ])
};
