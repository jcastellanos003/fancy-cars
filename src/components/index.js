export * from './ActiveLink/ActiveLink';
export * from './BackButton/BackButton';
export * from './Card';
export * from './CompareButton/CompareButton';
export * from './CompareLink/CompareLink';
export * from './Chip/Chip';
export * from './Header/Header';
export * from './HorizontalPane/HorizontalPane';
export * from './InnerPage/InnerPage';
export * from './Master/Master';
export * from './Meta/Meta';
export * from './Nav/Nav';
export * from './PageHeader/PageHeader';
export * from './PageTitle/PageTitle';
export * from './PriceLabel/PriceLabel';
export * from './SearchBox/SearchBox';
export * from './SelectedCard/SelectedCard';
export * from './SelectedButton/SelectedButton';
export * from './VehicleCard/VehicleCard';
