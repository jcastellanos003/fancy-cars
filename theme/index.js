export * from './DefaultTheme';
export * from './GlobalStyles';
