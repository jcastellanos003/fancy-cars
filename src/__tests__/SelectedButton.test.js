import React from 'react';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components'

import { SelectedButton } from '../components';
import { theme } from '../theme';

describe('SelectedButton Component', () => {
  test('Matches the snapshot', () => {
    const handleClick = event => event.preventDefault();
    const id = 1;
    const text = 'Selected test';
    const selectedButton = create(
      <ThemeProvider theme={theme}>
        <SelectedButton id={id} text={text} onClick={handleClick} />
      </ThemeProvider>
      );

    expect(selectedButton.toJSON()).toMatchSnapshot();
  })
})
