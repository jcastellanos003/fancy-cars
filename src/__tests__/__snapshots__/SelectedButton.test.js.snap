// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedButton Component Matches the snapshot 1`] = `
.c0 {
  border-radius: 5px;
  display: block;
  text-align: center;
  cursor: pointer;
  background: transparent;
  color: #FAFAFA;
  border: 2px solid #FFC82C;
}

.c0:hover {
  background: #FFC82C;
  color: #2A293B;
}

.c0 span {
  display: block;
  padding: 0.2rem 0.5rem;
}

<span
  className="c0"
>
  <span
    item={1}
    onClick={[Function]}
  >
    Selected test
  </span>
</span>
`;
